### PR TITLE
Github action: Inline lint python / js / cpp

### PR DESCRIPTION
**Pull request by justinchuby**
**Description**: Uses the reviewdog action which supports inline reporting. https://github.com/reviewdog/action-black 

Replaces #11326 

See demo: #11337

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by justinchuby** [comment]
It will show inline annotations like so: 

img

https://github.com/reviewdog/reviewdog

**Comment by justinchuby** [comment]
Demo: #11335

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          github_token: ${{ secrets.github_token }}
+          # Change reviewdog reporter if you need [github-pr-check, github-check, github-pr-review].
+          reporter: github-pr-check
+          # Change reporter level if you need.
+          # GitHub Status Check won't become failure with a warning.
+          level: error
+          filter_mode: file
+      - uses: reviewdog/action-flake8@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tianleiwu) [inline .github/workflows/lint.yml:27]: nit: lower case ("github_token") and upper case ("GITHUB_TOKEN") are used in different places. We can use lower case for consistency.

Comment (justinchuby) [reply .github/workflows/lint.yml]: Done

**Review by tianleiwu — APPROVED** [review]

**Pull request merged**